src/db/mongo.ts

```diff
@@ -182,19 +185,15 @@ export class ProsopoDatabase implements Database {
         if (!isHex(treeRoot)) {
             throw new Error(`${ERRORS.DATABASE.INVALID_HASH.message}: treeRoot`)
         }
-        // create a bulk upsert operation and execute
+        // create a bulk ceate operation and execute
         await this.tables.solutions?.bulkWrite(captchas.map((captchaDoc) => ({
-            updateOne: {
-                filter: { _id: captchaDoc.captchaId },
-                update: {
-                    $set:
-                        {
-                            solution: captchaDoc.solution,
-                            salt: captchaDoc.salt,
-                            treeRoot
-                        }
-                },
-                upsert: true
+            insertOne: {
```

Comment:
just out of curiosity, why we'd move from upsert to insert only?

Reply:
Actually update is overriding the solutions on the basis of captchaId. So at the end we're getting only one solution of a particular captcha. So to overcome this issue I've used insertOne to record all the solutions provided by the users for a particular captcha. Hope it make sense.

Reply:
sounds good to me, cc @forgetso